pkg/network/tracer/connection/failure/matching.go

```diff
@@ -110,16 +110,23 @@ func (fc *FailedConns) MatchFailedConn(conn *network.ConnectionStats) {
 	util.ConnStatsToTuple(conn, fc.failureTuple)
 
 	fc.RLock()
-	defer fc.RUnlock()
+	foundMatch := false
 
 	if failedConn, ok := fc.FailedConnMap[*fc.failureTuple]; ok {
 		// found matching failed connection
-		conn.TCPFailures = make(map[uint32]uint32)
+		foundMatch = true
+		conn.TCPFailures = failedConn.CountByErrCode
 
-		for errCode, count := range failedConn.CountByErrCode {
-			failureTelemetry.failedConnMatches.Add(1, strconv.Itoa(int(errCode)))
-			conn.TCPFailures[errCode] += count
+		for errCode := range failedConn.CountByErrCode {
+			failureTelemetry.failedConnMatches.Add(1, unix.ErrnoName(syscall.Errno(errCode)))
 		}
+		failedConn.CountByErrCode = nil
+	}
+	fc.RUnlock()
+	if foundMatch {
```

Comment:
Why not just delete it above?

Reply:
Only the read lock is held above.

Reply:
I don't have a perfect understanding of RXMutex, but I was wary about trying to acquire a write lock inside of a read lock

Reply:
You cannot acquire a write lock while a read lock is held.
> A RWMutex is a reader/writer mutual exclusion lock. The lock can be held by an arbitrary number of readers or a single writer.

Reply:
I think you may need to use the write lock here. In theory you could have two callers come through with a match on the failures and would have the same failures attached to two connections.

Reply:
that's a fair edge case... I'll just apply a write lock for the entire block